pkg/hubble/parser/threefour/parser.go

```diff
@@ -287,13 +293,41 @@ func (p *Parser) resolveNames(epID uint32, ip netip.Addr) (names []string) {
 	return nil
 }
 
-func decodeLayers(packet *packet) (
+func decodeLayers(payload []byte, packet *packet, isL3Device, isIPv6, isVXLAN, isGeneve bool) (
 	ethernet *pb.Ethernet,
 	ip *pb.IP,
 	l4 *pb.Layer4,
+	tunnel *pb.Tunnel,
 	sourceIP, destinationIP netip.Addr,
 	sourcePort, destinationPort uint16,
-	summary string) {
+	summary string,
+	err error,
+) {
+	// Since v1.1.18, DecodeLayers returns a non-nil error for an empty packet, see
+	// https://github.com/google/gopacket/issues/846
+	// TODO: reconsider this check if the issue is fixed upstream
+	if len(payload) == 0 {
+		return
+	}
+
+	packet.Lock()
+	defer packet.Unlock()
+
+	// Truncate layers to avoid accidental re-use.
+	packet.Layers = packet.Layers[:0]
```

Comment:
nit: I'm not entirely sure I understand why we do this truncation. Previously, we only ever truncated the layers when the payload is empty, now we unconditionally do it when it's not empty. AFAICT the decoder will truncate it anyways.

Reply:
Yep, I believe this line should go above:

```go
if len(payload) == 0 {
	  packet.Layers = packet.Layers[:0]
		return
	}
```